### PR TITLE
Generalize PID term descriptions

### PR DESCRIPTION
**Pull request by Hwurzburg**
NFC....allows application to motors, tilts, control surfaces, etc.

confused a user

### CODE REVIEW

**Review by TunaLobster — APPROVED** [review]

**Review by lthall — COMMENTED** [review]
These comments are not correct. I don't think it is a good idea to change a comment so a person who doesn't understand control principles is not confused, to then confuse all the people that do by making it incorrect.

Depending on the aircraft these outputs go to motor drive voltage or actuator position / angle. If you wanted to go up one level of abstraction you could say commanded rotational torque or rate (but generally a combination of both to varying degrees, depending on the actuator).

You could change it to actuator output but even this isn't correct as many frames go through a liniearisateion between the controller output and actuator output.

Sorry but I can't think of a change here that I would approve. Happy to hear other suggestions though.

**Comment by Hwurzburg** [comment]
Present description is incorrect and misleading....a change must be made

**Comment by tridge** [comment]
@lthall could you suggest a wording that doesn't tie it to motor speeds? It's wrong saying it is motor speed for lots of vehicles (tri-copter tail, helicopter swash, quadplane tilt yaw, steering on most rovers).

**Comment by lthall** [comment]
> Present description is incorrect and misleading....a change must be made

Ahh, is that so. Maybe start with explaining how you believe it is incorrect. I suspect there is a missunderstanding about how this works.

And for future reference, changes to copter control loop discriptions should list me a reviewer.


> @lthall could you suggest a wording that doesn't tie it to motor speeds? It's wrong saying it is motor speed for lots of vehicles (tri-copter tail, helicopter swash, quadplane tilt yaw, steering on most rovers).

For multirotors the closest actuator discription would be roll, pitch or yaw torque. It has the motor mixer -> thrust liniearisation -> multiple PWM -> multiple Voltage Drive -> multiple Motor Efficiency and Load -> multiple Motor speeds. So discribing it as motor speed is far from accurate.
For helicopters it is swash plate angle for roll and ptich and blade pitch or thrust for yaw. (There will eventually be linearisation in there too probably). No speed here at all.
For rovers it is steering angle motor drive or differential motor drive. Maybe with encoders or a governer there may be one output here that actually is speed.

So to lable this as a "speed" output ranges from completly inaccurate to completly irelevent.

**Comment by lthall** [comment]
If the P term somehow needs fixing why don't the D, I and I max terms also need to be fixed.

Maybe the way to go is to use a similar discription to I and D. Something like "Corrects proportional to the difference in desired roll rate vs actual roll rate"

**Comment by Hwurzburg** [comment]
changed P to @lthall suggestion....also cleaned up IMAXs

**Comment by lthall** [comment]
Oh I am a complete retard. I thought you were changing it TOO "motor speed". My absolute and syncerest apologies!!!!

Now I undestand why you felt justified in saying "Present description is incorrect and misleading....a change must be made". I didn't realise but I was actually firmly defending making the change!

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ const AP_Param::GroupInfo AC_AttitudeControl_Heli::var_info[] = {
 
     // @Param: RAT_RLL_P
     // @DisplayName: Roll axis rate controller P gain
-    // @Description: Roll axis rate controller P gain.  Converts the difference between desired roll rate and actual roll rate into a motor speed output
+    // @Description: Roll axis rate controller P gain.  Corrects in proportion to the difference between the desired roll rate vs actual roll rate output
```
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp:20]: Corrects in proportion to the difference between the desired roll rate vs actual roll rate

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp [hunk 2]
```diff
@@ -91,7 +91,7 @@ const AP_Param::GroupInfo AC_AttitudeControl_Heli::var_info[] = {
 
     // @Param: RAT_PIT_P
     // @DisplayName: Pitch axis rate controller P gain
-    // @Description: Pitch axis rate controller P gain.  Converts the difference between desired pitch rate and actual pitch rate into a motor speed output
+    // @Description: Pitch axis rate controller P gain.  Corrects in proportion to the difference between the desired pitch rate vs actual pitch ra
```
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp:94]: Corrects in proportion to the difference between the desired pitch rate vs actual pitch

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp [hunk 3]
```diff
@@ -77,7 +77,7 @@ const AP_Param::GroupInfo AC_AttitudeControl_Multi::var_info[] = {
 
     // @Param: RAT_PIT_P
     // @DisplayName: Pitch axis rate controller P gain
-    // @Description: Pitch axis rate controller P gain.  Converts the difference between desired pitch rate and actual pitch rate into a motor speed output
+    // @Description: Pitch axis rate controller P gain.  Corrects in proportion to the difference between the desired pitch rate vs actual pitch rate output
```
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp:80]: Corrects in proportion to the difference between the desired pitch rate vs actual pitch rate

**Review by lthall — CHANGES_REQUESTED** [review]
Happy to approve this with the little mistakes fixed.

Again I apologise for mixing up the red and green highlighting and making an ass of myself.

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp (hunk 2 above)
Comment (Hwurzburg) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp:94]: fixed

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp:20]: copy paste error...fixed

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp (hunk 3 above)
Comment (Hwurzburg) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp:80]: fixed

**Comment by Hwurzburg** [comment]
@lthall no problem, thanks for reviewing

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp [hunk 4]
```diff
@@ -9,7 +9,7 @@ const AP_Param::GroupInfo AC_AttitudeControl_Multi::var_info[] = {
 
     // @Param: RAT_RLL_P
     // @DisplayName: Roll axis rate controller P gain
-    // @Description: Roll axis rate controller P gain.  Converts the difference between desired roll rate and actual roll rate into a motor speed output
+    // @Description: Roll axis rate controller P gain.  CCorrects in proportion to the difference between the desired roll rate vs actual roll rate
```
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp:12]: Little typo.. a double "C" here..

**Review by lthall — APPROVED** [review]
Aprove with Randy's fix.

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp (hunk 4 above)
Comment (Hwurzburg) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Multi.cpp]: done

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**